src/solvers/flattening/boolbv_array_of.cpp

```diff
@@ -56,7 +59,8 @@ bvt boolbvt::convert_array_of(const array_of_exprt &expr)
       bv[offset]=tmp[j];
   }
 
-  assert(offset==bv.size());
+  INVARIANT(
+    offset == bv.size(), "we shall have iterated over all the bit indices");
```

Comment:
How about rewriting the above nested loops so that the iteration is actually: `for(auto &b : bv)` with a counter `j` that is reset to `0` each time it reaches `tmp.size()`, or possibly even using an iterator over `tmp`? Let's make the code trivially guarantee the invariants, not requiring sanity checking after the fact.